boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,943 @@
+     TRACK_AND_HOLD_CURRENT_DISABLE | WATCHDOG_DISABLE |                     \
+     VDD_FAILURE_DETECTION_ENABLE | CSNS_FUNCTION_DISABLED |                 \
+     OVERVOLTAGE_PROTECTION_ENABLE)
+
+// Channel  Configurations
+#define RETRY_CONFIG                                               \
+    (RANDOM_CURRENT_OFFSET_SUBTRACT | AUTO_RETRY_NUMBER_INFINITE | \
+     AUTO_RETRY_PERIOD_T_AUTO_10 | AUTO_RETRY_DC_MOTOR_DISABLED)
+#define CONFR_CONFIG                                                  \
+    (SHORT_CIRCUIT_DETECTION_ENABLE | OPEN_LOAD_DETECTION_ON_ENABLE | \
+     OPEN_LOAD_DETECTION_OFF_ENABLE | DIRECT_CONTROL_ENABLE |         \
+     SWITCH_ON_DELAY_NO_DELAY | SLEW_RATE_MEDIUM)
+#define OCR_LOW_CURRENT_SENSE_CONFIG                              \
+    (PWM_EXTERNAL_CLOCK_DIVIDER_256 | PWM_CLOCK_SOURCE_EXTERNAL | \
+     THRESHOLD_ACTIVATION_TIMES_T_OCH1_T_OCM1_L |                 \
+     MEDIUM_CURRENT_THRESHOLD_SELECT_IOCM2 |                      \
+     LOW_CURRENT_THRESHOLD_SELECT_IOCL2 |                         \
+     HIGH_CURRENT_THRESHOLD_SELECT_IOCH2 | CURRENT_SENSE_RATIO_LOW)
+#define OCR_HIGH_CURRENT_SENSE_CONFIG                             \
+    (PWM_EXTERNAL_CLOCK_DIVIDER_256 | PWM_CLOCK_SOURCE_EXTERNAL | \
+     THRESHOLD_ACTIVATION_TIMES_T_OCH1_T_OCM1_L |                 \
+     MEDIUM_CURRENT_THRESHOLD_SELECT_IOCM2 |                      \
+     LOW_CURRENT_THRESHOLD_SELECT_IOCL3 |                         \
+     HIGH_CURRENT_THRESHOLD_SELECT_IOCH2 | CURRENT_SENSE_RATIO_HIGH)
+
+struct Efuse_Context
+{
+    // The SPI handle for the SPI device the E-Fuse is connected to
+    SPI_HandleTypeDef *efuse_spi_handle;
+
+    // The current state of the watchdog-in bit (bit 15). If the watchdog is
+    // enabled its state must be alternated at least once within the watchdog
+    // timeout period.
+    bool wdin_bit_to_set;
+};
+
+/**
+ * Set the current/temperature monitoring option of CSNS pin
+ * @param selection The desired monitoring function which can be selected by
+ * specifying the following values:
+ *                  CSNS_FUNCTION_DISABLED - Current sensing disabled
+ *                  AUX1_CURRENT_SENSE_CHANNEL - Current sensing for Aux 1
+ *                  AUX2_CURRENT_SENSE_CHANNEL - Current sensing for Aux 2
+ *                  CSNS_FUNCTION_TEMPERATURE - Efuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed
+ * channels (for Parallel mode)
+ * @return EXIT_CODE_OK if the channel configuration was successful
+ *         EXIT_CODE_TIMEOUT if one of the SPI writes timed-out
+ */
+static ExitCode Io_Efuse_Aux1Aux2ConfigureChannelMonitoring(uint8_t selection);
+
+/**
+ * Exit fail-safe mode and disable the watchdog timer
+ * @return EXIT_CODE_OK if the efuse exited fail-safe mode
+ *         EXIT_CODE_TIMEOUT if the one of the SPI writes timed-out
+ *         EXIT_CODE_UNIMPLEMENTED if the efuse failed to exit fail-safe mode
```

Comment:
I used _EXIT_CODE_UNIMPLEMENTED_ here because I couldn't really think of a enum name. We should change this to a new ExitCode Enum

Reply:
Preferably something not too specific so it could be used possibly by another board. ie.
`EXIT_CODE_FAIL_SAFE_MODE` would be a bad name lol